server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -802,6 +806,46 @@ static int getIndexNumberOfRoutingShards(Settings indexSettings, @Nullable Index
         return resolvedAliases;
     }
 
+    /**
+     * Validate and resolve the aliases explicitly set for the index, together with the ones inherited from the specified
+     * templates.
+     *
+     * The template mappings are applied in the order they are encountered in the list (clients should make sure the lower index, closer
+     * to the head of the list, templates have the highest {@link IndexTemplateMetadata#order()})
+     *
+     * @return the list of resolved aliases, with the explicitly provided aliases occurring first (having a higher priority) followed by
+     * the ones inherited from the templates
+     */
+    public static List<AliasMetadata> resolveAndValidateAliases(String index, Set<Alias> aliases,
```

Comment:
I think the additional logic of this method at line 827-828 and 837 to 841 can be inlined into the existing `resolveAndValidateAliases(...)` method?

Reply:
This was a step in the right direction. Removed the overload and inlined the expression resolver